src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts

```diff
@@ -0,0 +1,75 @@
+import { useWalletInfo } from '@cow/modules/wallet'
+import { Token } from '@uniswap/sdk-core'
+import { atom, useAtom, useAtomValue } from 'jotai'
+import { loadable } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+const BASE_URL = 'https://temp.api.uniswap.org/v1/tokens/search?tokenQuery='
+
+type Address = `0x${string}`
+
+interface FetchTokensResult {
+  chainId: number
+  name: string
+  address: Address
+  decimals: number
+  symbol: string
+  logoURI: string
+  coinGeckoId: string
+  priceUsd: number
+  price24hChange: number
+  volume24h: number
+  marketCap: number
+}
+
+interface FetchTokensApiResult {
+  code: number
+  status: 'success' | 'failed'
+  hasNext: boolean
+  data: FetchTokensResult[]
+}
+
+async function fetchTokens(query: string): Promise<FetchTokensApiResult> {
+  const response = await fetch(`${BASE_URL}${query}`)
+
+  return await response.json()
+}
+
+const searchQuery = atom('')
+
+const tokensData = loadable(
+  atom(async (get) => {
+    const query = get(searchQuery)
+
+    if (query.length > 0) {
+      return await fetchTokens(query)
+    }
+
+    return undefined
+  })
+)
```

Comment:
This is state. Normally we place these ones in a module's state.

Just curious, why we use joitai state here? 
what happen if we use this hook in two different parts of the UI? that useEffect will trigger the setQuery and they will be fighting for the right to update the atom?